common/src/main/java/com/microsoft/hydralab/common/util/AttachmentService.java

```diff
@@ -46,25 +47,26 @@ public class AttachmentService {
     public StorageFileInfo addAttachment(String entityId, EntityType entityType, StorageFileInfo storageFileInfo, File file, Logger logger) {
         boolean recordExists = false;
 
-        storageFileInfo.setBlobContainer(entityType.storageContainer);
+        storageFileInfo.setBlobContainer(entityType.getStorageContainer());
         List<StorageFileInfo> tempFileInfos = storageFileInfoRepository.queryStorageFileInfoByMd5(storageFileInfo.getMd5());
+        StorageFileInfo tmp = storageFileInfo;
         for (StorageFileInfo tempFileInfo : tempFileInfos) {
             if (compareFileInfo(storageFileInfo, tempFileInfo)) {
-                storageFileInfo = updateFileInStorageAndDB(tempFileInfo, file, entityType, logger);
+                tmp = updateFileInStorageAndDB(tempFileInfo, file, entityType, logger);
                 recordExists = true;
                 break;
             }
         }
         if (!recordExists) {
-            storageFileInfo = saveFileInStorageAndDB(storageFileInfo, file, entityType, logger);
+            tmp = saveFileInStorageAndDB(storageFileInfo, file, entityType, logger);
         }
-        saveRelation(entityId, entityType, storageFileInfo);
+        saveRelation(entityId, entityType, tmp);
         file.delete();
         return storageFileInfo;
```

Comment:
The returned storageFileInfo should be enriched by either updateFileInStorageAndDB or saveFileInStorageAndDB.
What issue are the changes aiming at resolving?

Reply:
We don't want to reassign param in function(storageFileInfo for example), so we should return tmp here since tmp will be enriched.